tests/unit/utils/test_osv.py

```diff
@@ -0,0 +1,415 @@
+                {
+                    "package": {"name": "test-pkg", "ecosystem": "test-eco"},
+                    "ranges": [
+                        {
+                            "type": "ECOSYSTEM",
+                            "events": [
+                                {"introduced": "1.0.0"},  # No fixed field
+                                {"fixed": ""},  # Empty fixed field
+                            ]
+                        }
+                    ]
+                }
+            ]
+        }
+        fixdater = Mock()
+
+        osv.patch_fix_date(advisory, fixdater)
+
+        fixdater.find.assert_not_called()
+
+    def test_fixdater_no_results_no_changes(self):
+        """Test that when fixdater returns no results, no changes are made."""
+        advisory = {
+            "id": "test-vuln",
+            "affected": [
+                {
+                    "package": {"name": "test-pkg", "ecosystem": "test-eco"},
+                    "ranges": [
+                        {
+                            "type": "ECOSYSTEM",
+                            "events": [{"fixed": "1.0.1"}]
+                        }
+                    ]
+                }
+            ]
+        }
+        fixdater = Mock()
+        fixdater.find.return_value = []  # No results
+
+        osv.patch_fix_date(advisory, fixdater)
+
+        # Should have called find but made no changes to database_specific
+        fixdater.find.assert_called_once_with(
+            vuln_id="test-vuln",
+            cpe_or_package="test-pkg",
+            fix_version="1.0.1",
+            ecosystem="test-eco"
+        )
+
+        # No database_specific should be added
+        assert "database_specific" not in advisory["affected"][0]["ranges"][0]
+
+    def test_successful_fix_date_patching(self):
+        """Test successful patching of fix date data."""
+        advisory = {
+            "id": "CVE-2023-1234",
+            "affected": [
+                {
+                    "package": {"name": "example-pkg", "ecosystem": "npm"},
+                    "ranges": [
+                        {
+                            "type": "ECOSYSTEM",
+                            "events": [{"fixed": "2.1.0"}]
+                        }
+                    ]
+                }
+            ]
+        }
+
+        # Mock fixdater result
+        fix_result = Mock(spec=FixResult)
+        fix_result.date = datetime(2023, 6, 15, 10, 30, 45)
+        fix_result.kind = "release"
+
+        fixdater = Mock()
+        fixdater.find.return_value = [fix_result]
+
+        osv.patch_fix_date(advisory, fixdater)
+
+        # Verify the call
+        fixdater.find.assert_called_once_with(
+            vuln_id="CVE-2023-1234",
+            cpe_or_package="example-pkg",
+            fix_version="2.1.0",
+            ecosystem="npm"
+        )
+
+        # Verify the fix data was added
+        expected_fix = {
+            "version": "2.1.0",
+            "date": "2023-06-15T10:30:45",
```

Comment:
The test fixtures elsewhere have dates like `"2024-01-14"`. I'm just making sure nothing is expected to truncate this to only the date portion. Just following up on being surprised to see different date format here than in `tests/unit/providers/bitnami/test-fixtures/snapshots/bit-consul-2021-37219.json` above, for example.

Reply:
nice catch -- updated to just dates